controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -163,6 +164,13 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return reconcile.Result{}, err
 	}
 
+	// Verify odh-trusted-ca-bundle Configmap is created for all the namespaces
+	err = r.addTrustedCABundleConfigMap(ctx, instance)
```

Comment:
Maybe path worth taking is something different:

- when we start the controller do the bulk update of existing namespaces (matching some criteria)
- then reconcile only on creation of the new namespace

Reply:
The reason I added it here is, that we would need to reconcile the configmap if there are any changes to the `spec.trustedCABundle` field in DSCI.

Also note, that this is a short-term solution until we have an identifier for Data Science Project namespaces.

Reply:
That's why I put _even unrelated_ changes in my original comment :) I know that we need to update the bundle. It's intentional that we do not have access to old and new instances in the main reconcile func, so I am not suggesting hacking our way out of it.

What I was wondering is: 
- we know the previous value if it's stored in the cfgmap in e.g. appNs
- we can check if it's different now and only then trigger reconcile for all ns if needed.

> Also note, that this is a short-term solution until we have an identifier for Data Science Project namespaces.

I am well aware of this limitation :) But in Eastern Europe we have this saying: "Nothing is more permanent than a temporary solution" :) And I learned this the hard way.

So my line of thinking is to make sure it's as good as it can be considering the circumstances.